python/cudf/cudf/_lib/cpp/io/orc.pxd

```diff
@@ -69,7 +69,7 @@ cdef extern from "cudf/io/orc.hpp" \
         size_type get_stripe_size_rows() except +
         size_type get_row_index_stride() except +
         cudf_table_view.table_view get_table() except +
-        const cudf_io_types.table_input_metadata *get_metadata() except +
+        const cudf_io_types.table_input_metadata& get_metadata() except +
```

Comment:
AIUI, the C++ interface now returns a `std::optional<table_input_metadata>`, how does this work?


I _suspect_ it is because these Cython forward-declarations are only used for typing the transpilation and so this will (if it is used) cause some C++ to be generated that will then fail to compile.

As it happens, I think `get_metadata` is never called anywhere in the Cython codebase, so therefore things "work".

So maybe
```cython
from libcpp.optional cimport optional

...

const optional[cudf_io_types.table_input_metadata]& get_metadata() except +
```
?

Reply:
nice catch! Fixed all four getters.